### PR TITLE
Chore: remove unused code and fix typo

### PR DESCRIPTION
**Pull request by yyzxw**
### Summary
🐛🚀📝

This pull request fixes various typos, renames some variables, adds error handling, and improves comments and readability in several files. It also removes a redundant function and refactors the validation logic for the rollout plan. The changes affect the `pkg`, `charts`, and `references` directories.

> _Sing, O Muse, of the valiant pull request_
> _That fixed many typos and errors in code,_
> _And made the functions and variables more clear_
> _With consistent names and comments as guide._

### Walkthrough
*  Add comments to explain the meaning of parameters in the `values.yaml` file for the vela-core chart ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-5de38b74432257fd0da10f3a511568790f1856d14e3385d82d2f0b8b72f6163fL16-R16), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-5de38b74432257fd0da10f3a511568790f1856d14e3385d82d2f0b8b72f6163fL23-R23), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-5de38b74432257fd0da10f3a511568790f1856d14e3385d82d2f0b8b72f6163fL29-R29))
*  Remove redundant parameter names from comments in the `values.yaml` file for the vela-core chart ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-5de38b74432257fd0da10f3a511568790f1856d14e3385d82d2f0b8b72f6163fL229-R232))
*  Fix typos in the `pkg/addon/reader_gitee.go` and `pkg/appfile/parser.go` files ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-f72634c9a59d7c655ace07efa3f7689081e1f74d860ced90a90e8119b084c461L103-R103), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-72491e6e5c2c0f6891265e6a4a636ff88e7db16e330f8c913873bd69b190a803L121-R121), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-72491e6e5c2c0f6891265e6a4a636ff88e7db16e330f8c913873bd69b190a803L179-R179), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-72491e6e5c2c0f6891265e6a4a636ff88e7db16e330f8c913873bd69b190a803L263-R263))
*  Rename variables to avoid name collisions with imported packages in the `pkg/config/provider/provider.go`, `pkg/controller/core.oam.dev/v1alpha2/application/revision.go`, `pkg/cue/script/template.go`, `pkg/utils/env/env.go`, and `references/cli/addon.go` files ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-424a90de3b5dc1c029197c7dfefd68596be96dbca82720b01025e45dd2aac022L117-R122), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-b76130d531d3dc85bc60130d337c8807ee0a27ad095ed44c0284dc34545432b6L785-R787), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-b76130d531d3dc85bc60130d337c8807ee0a27ad095ed44c0284dc34545432b6L797-R797), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-b76130d531d3dc85bc60130d337c8807ee0a27ad095ed44c0284dc34545432b6L808-R810), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-dcc0ec36c20763b3796ca20afbe15509650407540d12a2672b1b11084851987fL223-R230), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-dfaf26b5c5c183fe52db56fa1233a24bcacac7c71668d62beff433f4ec9f5cd6L278-R287), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-5c1ce01bb88762ec6a5abb96531e8ac946cff33a1fa7ff2e033131f8dcf0e2b8L1161-R1167))
*  Add pointer receivers to methods of the `ossReader` type in the `pkg/addon/reader_oss.go` file ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-74ef6cab5623fe8750968d3816d9d738372e342d39b230a806b2a9b006cc6d79L103-R103))
*  Remove unnecessary byte conversions in the `pkg/controller/utils/capability.go` and `pkg/cue/script/template_test.go` files ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-84088786897ac51a95b85eeab65171013b54c6d4b53edf295fb457de4e62becdL750-R750), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-7d7eeaed1ef67dcf51a0e6a354bf4e8cb19008c28b4aec3191f4e5ac1fd22221L291-R296))
*  Replace `:=` operator with `var` keyword to declare slices in the `pkg/controller/utils/capability.go` and `pkg/velaql/providers/query/collector.go` files ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-84088786897ac51a95b85eeab65171013b54c6d4b53edf295fb457de4e62becdL334-R334), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-fcbbd032b71e076ff83ed5f3c7086a3a84281185813a6f08ed0742cbd56655cdL210-R210))
*  Add error handling to methods that return errors in the `pkg/cue/process/handle_test.go`, `pkg/cue/task/process_test.go`, and `pkg/webhook/common/rollout/rollout_plan.go` files ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-140cf4fcf0a14a752e2c44cce7a7431d592a64836570f3ba84dd8e518aa6de4dL106-R120), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-cdc40a856d2fa84cd855d35aa2d2c20470e79293eac05c13d54e2ffb4761bc94L91-R91), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-cdc40a856d2fa84cd855d35aa2d2c20470e79293eac05c13d54e2ffb4761bc94L99-R99), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-d87c99ea9752161d7b69c7ab0528c8a8f1046605eb1d83bf85bea69f7b003118L155-L163))
*  Remove the `ValidateUpdate` function from the `pkg/webhook/common/rollout/rollout_plan.go` file as part of a larger refactor ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-d87c99ea9752161d7b69c7ab0528c8a8f1046605eb1d83bf85bea69f7b003118L155-L163))
*  Fix a typo in the comment for the `SliceIncludeSlice` function in the `pkg/utils/strings.go` file ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-9128bfe925e886e4891edaa9667c047ea3cc83b9b1e2968e0c01c9bec19201d6L71-R71))
*  Remove the redundant parameter name from the comment for the `Sensitive` field of the `Template` type in the `pkg/config/factory.go` file ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-a7bb21919ecd95689199a6a4f1b1d013c8dbe734306be14b2a916f2a21e1b1e2L121-R121))
*  Add a comma to improve the grammar of the `Context` section of the `pkg/monitor/README.md` file ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-b5361dd3bddc84b6ab9b9950ec1cbcad22e2366987209e0f00fe8249f289b211L5-R5))
*  Rename variables to make them more descriptive and consistent in the `references/cli/cluster.go` and `references/cli/config.go` files ([link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-9c055c6e7803231f2425e1d77339675716976bcd53e2801fd60ce606bb4fadb2L113-R117), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-11d168ea30a954837432bef072cd9af25a9b679e9e99fff758a43ea56d4ff952L147-R147), [link](https://github.com/kubevela/kubevela/pull/5986/files?diff=unified&w=0#diff-11d168ea30a954837432bef072cd9af25a9b679e9e99fff758a43ea56d4ff952L157-R157))

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/reader_gitee.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (g *giteeReader) listAddonMeta(dirPath string) ([]Item, error) {
 	return res, nil
 }
 
-// ReadFile read file content from github
+// ReadFile read file content from gitee
 func (g *giteeReader) ReadFile(relativePath string) (content string, err error) {
```
Comment (Somefive) [inline pkg/addon/reader_gitee.go:104]: Why change this?

File: pkg/controller/utils/capability.go [hunk 2]
```diff
@@ -331,7 +331,7 @@ func generateJSONSchemaWithRequiredProperty(schemas map[string]*openapi3.Schema,
 
 // GetKubeSchematicOpenAPISchema gets OpenAPI v3 schema based on kube schematic parameters for component and trait definition
 func GetKubeSchematicOpenAPISchema(params []commontypes.KubeParameter) ([]byte, error) {
-	required := []string{}
```
Comment (Somefive) [inline pkg/controller/utils/capability.go:334]: Let's keep this for empty array case (not let empty array to be nil).

File: pkg/cue/process/handle_test.go [hunk 3]
```diff
@@ -103,9 +103,21 @@ image: "myserver"
 		WorkflowName:    "myworkflow",
 		PublishVersion:  "mypublishversion",
 	})
-	ctx.SetBase(base)
```
Comment (Somefive) [inline pkg/cue/process/handle_test.go:106]: We do not need to handle so many error in unit test as they will not be triggered.

File: pkg/velaql/providers/query/collector.go [hunk 4]
```diff
@@ -207,7 +207,7 @@ func (c *AppCollector) FindResourceFromResourceTrackerSpec(ctx context.Context,
 		klog.Errorf("query the resourcetrackers failure %s", err.Error())
 		return nil, err
 	}
-	var resources = []Resource{}
```
Comment (Somefive) [inline pkg/velaql/providers/query/collector.go:210]: Same as above

File: references/cli/addon.go [hunk 5]
```diff
@@ -1158,13 +1158,13 @@ func hasAddon(addons []*pkgaddon.UIData, name string) bool {
 	return false
 }
 
-func transClusters(cstr string) []interface{} {
```
Comment (Somefive) [inline references/cli/addon.go:1161]: I think cstr is better here as it refers to string

**Review by Somefive — COMMENTED** [review]
Generally LGTM after nit.

File: pkg/addon/reader_gitee.go (hunk 1 above)
Comment (yyzxw) [reply pkg/addon/reader_gitee.go:104]: from the file name `reader_gitee.go` this is to get gitee file, not github

File: pkg/controller/utils/capability.go (hunk 2 above)
Comment (yyzxw) [reply pkg/controller/utils/capability.go:334]: ye, i will revert it

File: pkg/addon/reader_gitee.go (hunk 1 above)
Comment (Somefive) [reply pkg/addon/reader_gitee.go:104]: ok, you are right :)

**Review by Somefive — APPROVED** [review]

**Comment by yyzxw** [comment]
@ charlie0129 @wangyikewxgm @wonderflow  
hi, I am very glad to contribute to kubevela. Could you please help me review this pr? Thank you very much.  best regards

**Review by wangyikewxgm — APPROVED** [review]

**Pull request merged**